### ISSUE TITLE
Union of interface with self intersected with additional properties does not throw error

### ISSUE THREAD

**Issue by WesCossick** [microsoft/TypeScript#37503]
**TypeScript Version:**  3.9.0-dev.20200319

**Search Terms:** "union intersection self"

**Code**

```ts
interface Vehicle {
    make: string;
    model: string;
}

interface Car extends Vehicle {
    isHatchback: boolean;
}

interface Truck extends Vehicle {
    bedLength: number;
}

interface ElectricProperties {
    isElectric: true,
    batterySize?: number;
    hydrogenTankSize?: number;
}

const car: Car | (Car & ElectricProperties) = {
    make: 'Toyota',
    model: 'Corolla',
    isHatchback: false,
    hydrogenTankSize: 123,
}
```

**Expected behavior:** I would expect TypeScript to complain that `car` can't use `hydrogenTankSize` unless `isElectric` was provided and set to `true`.

**Actual behavior:** No error is presented.

**Playground Link:** https://www.typescriptlang.org/play/?ts=3.9.0-dev.20200319&ssl=1&ssc=1&pln=25&pc=2#code/JYOwLgpgTgZghgYwgAgGoQBbAQGxQbwChkTkBbOAawgC5kBnMKUAcwG5jSyB7AEwhx1GzEO0IBfQoVCRYiFAGE4UZBAAekEL3ppM2PMiKlkwegAk4YBBgBGiSnRvdueOCA6Tp4aPCTIAKlAArgiUqhoQWjroWLgEnCQ2ELwAMpEsYBh0IEFkSVAeUjI+8sgAongITNgAClDcAA7QYMAQOkakphUQVcwIdExBEAA0Cch2YLIAngDKwABeEAD82bn5HMYYU7z1LJH+bpRziyvIOXnQhYQI3CCMyAjKdEoqAD7IABQvyABk5ZXVBB1RrNVr0ACUyAAvIYxhRqHQAOT+bhTbhgOCI0bGHj8QTIREKbj1HA4THYzrmSzWOyhOjwHD0EZjLY7bh7EAHEBHBa0ZAARgATABmUbiIA

**Related Issues:** I was not able to find similar issues.

**Comment by RyanCavanaugh** [comment]
The provided value is a valid subtype of `Car`, so it would be incorrect to error on this program. Excess property checking is mostly heuristic-based and this is a case that it considers to be valid based on how other people write types.

In general you should never write `T | SomeSubtypeOfT`; these types don't do what you want. You could write something like
```ts
type GasCar = Car & { isElectric?: false };
```
and then say `const Car = GasCar | (Car & ElectricProperties }`

**Comment by WesCossick** [comment]
Okay, that makes sense, I did not realize that. Thanks @RyanCavanaugh for your help!

For reference, @pbelbin suggested using `never` for properties that are electric only. Here's [his solution](https://www.typescriptlang.org/play/?ts=3.9.0-dev.20200319&ssl=1&ssc=1&pln=31&pc=2#code/JYOwLgpgTgZghgYwgAgGoQBbAQGxQbwChkTkBbOAawgC5kBnMKUAcwG5jSyB7AEwhx1GzEO0IBfQoVCRYiFAGE4UZBAAekEL3ppM2PMiKlkwegAk4YBBgBGiSnRvdueOCA6Tp4aPCTIAKlAArgiUqhoQWjroWLgEnCQ2ELwAMpEsYBh0IEFkSVAeUjI+8sgActwgAKJ4CEzYAApQ3AAO0GDAEDpGpKY1EHXMCAD82RAAbtAcxnZgsgCeAMrAAF4Qo8ggE1MJyBjzvM0skf5ulMtrG1uTBRJF3nJ+-YONzW1QHV2Gu3219Qh0JhBCDTUizBYXdbZXL5Xb7Q7cY4gU4gc6rKGbGE7TwISqMZAIZR0AAUShUADJypVnv8mq12p16ABKZAAH2QpOUyEpNKGdPen2ZyAAvN9jBRqHQAOT+bjzbhgOBSgA0ux4-EEyClCm4zRwOCVquMpgsVls9jo8Bw9AgRtI8KOJzOkLoAEYAEwAZlV4iAA).
